core/src/main/java/hudson/model/ListView.java

```diff
@@ -171,6 +176,9 @@ public boolean hasJobFilterExtensions() {
         return columns;
     }
 
+    public SortedSet<String> getJobNames() {
+        return jobNames;
```

Comment:
FindBugs will be likely grumbling about returning the modifiable collection. Wrapping it by https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#unmodifiableSet(java.util.Set) would make sense if you do not expect it  to be really patched

Reply:
Also, it defeats any synchronization attempts. Returning a copy might be an option

Reply:
Wrapped as asked for